The result of pointer arithmetic is of type ptrdiff_t, which is not
necessarily the same underlying type as ssize_t.

--- a/llvm/tools/llvm-objcopy/MachO/MachOWriter.cpp	2022-06-11 00:47:36.000000000 +1000
+++ b/llvm/tools/llvm-objcopy/MachO/MachOWriter.cpp	2022-06-16 23:41:23.000000000 +1000
@@ -566,7 +566,7 @@ void MachOWriter::writeCodeSignatureData
   uint8_t *CurrHashWritePosition = HashWriteStart;
   while (CurrHashReadPosition < HashReadEnd) {
     StringRef Block(reinterpret_cast<char *>(CurrHashReadPosition),
-                    std::min(HashReadEnd - CurrHashReadPosition,
+                    std::min(static_cast<ssize_t>(HashReadEnd - CurrHashReadPosition),
                              static_cast<ssize_t>(CodeSignature.BlockSize)));
     SHA256 Hasher;
     Hasher.update(Block);
